pilot/pkg/networking/util/util.go

```diff
@@ -435,6 +435,24 @@ func AddSubsetToMetadata(md *core.Metadata, subset string) {
 	}
 }
 
+// AddALPNOverrideToMetadata adds whether the the ALPN prefix should be added to the header
+// to the given core.Metadata struct, if metadata is not initialized, build a new metadata.
+func AddALPNOverrideToMetadata(metadata *core.Metadata, alpnOverride bool) {
+	if metadata == nil {
+		metadata = &core.Metadata{
+			FilterMetadata: map[string]*structpb.Struct{},
+		}
+	}
+
+	if istioMeta, ok := metadata.FilterMetadata[IstioMetadataKey]; ok {
```

Comment:
If `IstioMetadataKey` doesn't exist in the map, we probably want to create it right? As written, the alpn_override wouldn't be added

Reply:
Added